lib/rules/no-extra-parens.js

```diff
@@ -493,7 +521,14 @@ module.exports = {
             },
 
             BinaryExpression: checkBinaryLogical,
-            CallExpression: checkCallNew,
+
+            CallExpression(node) {
+                if (hasExcessParens(node) && node.parent.type === "MemberExpression") {
```

Comment:
This part of the change seems strange to me. I see that you added it in order to fix the `(foo()).bar` case, but I'm a bit concerned about having a special case for when the parent is a `MemberExpression`, because it's not clear to me why that case should be conceptually different. My worry is that this might be patching only the specific case that was brought up in https://github.com/eslint/eslint/issues/9156 without properly fixing the root cause, so there might be other cases of the same bug that this fix doesn't address.

(When writing programs that analyze source code, there is usually a much larger input space than can feasibly be covered with unit tests, so I want to make sure that there is a clear reason why it's correct to handle `MemberExpression` nodes differently here, aside from the fact that it makes a particular test pass.)

Looking at the structure of the rest of the `no-extra-parens` rule, it seems like most of the node listener methods verify parentheses around the *children* of the particular node, rather than the node itself. For example, the `checkCallNew` function checks parentheses around the `callee` of `CallExpression` and `NewExpression` nodes, rather than checking parentheses around the `CallExpression` and `NewExpression` nodes themselves. It seems like `node.parent` isn't used much in the rule at all.

Based on this, I would have expected the `(foo()).bar` case to get handled by the [`MemberExpression` listener](https://github.com/eslint/eslint/blob/7685fed33b15763ee3cf7dbe1facfc5ba85173f3/lib/rules/no-extra-parens.js#L576-L595), not the `CallExpression` listener. Do we know why the `MemberExpression` listener wasn't handling this case correctly?